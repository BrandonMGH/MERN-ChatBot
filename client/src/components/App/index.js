import React from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';

import {
  INIT, HOME, EXPERIENCE,
  EDUCATION, PROJECTS,
  FEEDBACK,
} from 'constants/routes';

import Layout from 'hoc/layout';

import Home from 'containers/HomePage';

import Disclaimer from '../Pages/Disclaimer';
import Education from '../Pages/Education';
import Experience from '../Pages/Experience';
import Projects from '../Pages/Projects';
import Feedback from '../Pages/Feedback';

const App = () => (
  <BrowserRouter>
    <Switch>
      <Route path={INIT} exact component={Disclaimer} />
      <Layout>
        <Route path={HOME} exact component={() => <Home />} />
        <Route path={EXPERIENCE} exact component={Experience} />
        <Route path={EDUCATION} exact component={Education} />
        <Route path={PROJECTS} exact component={Projects} />
        <Route path={FEEDBACK} exact component={Feedback} />
      </Layout>
    </Switch>
  </BrowserRouter>
);

export default App;
