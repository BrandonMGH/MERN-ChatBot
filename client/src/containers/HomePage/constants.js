export const actionTypes = {
  DISCOVER_HOME_PAGE: 'HOME_PAGE/DISCOVER_HOME_PAGE',
  DISCOVER_NAME: 'HOME_PAGE/DISCOVER_NAME',
  DISCOVER_POSITION: 'HOME_PAGE/DISCOVER_POSITION',
  DISCOVER_LOCATION: 'HOME_PAGE/DISCOVER_LOCATION',
  DISCOVER_CONTACT_INFO: 'HOME_PAGE/DISCOVER_CONTACT_INFO',
  DISCOVER_AGE: 'HOME_PAGE/DISCOVER_AGE',
  DISCOVER_NATIONALITY: 'HOME_PAGE/DISCOVER_NATIONALITY',
  DISCOVER_ABOUT_ME: 'HOME_PAGE/DISCOVER_ABOUT_ME',
  DISCOVER_LANGUAGES: 'HOME_PAGE/DISCOVER_LANGUAGES',
  DISCOVER_HOBBIES: 'HOME_PAGE/DISCOVER_HOBBIES',
};
