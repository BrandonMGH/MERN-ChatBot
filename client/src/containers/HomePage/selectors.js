import { createSelector } from 'reselect';

const selectHomePageDomain = () => state => state.get('homePageDomain');

const selectDiscovered = () => createSelector(
  selectHomePageDomain(),
  homePageDomain => homePageDomain.get('discovered'),
);

const selectName = () => createSelector(
  selectHomePageDomain(),
  homePageDomain => homePageDomain.get('name'),
);

const selectPosition = () => createSelector(
  selectHomePageDomain(),
  homePageDomain => homePageDomain.get('position'),
);

const selectLocation = () => createSelector(
  selectHomePageDomain(),
  homePageDomain => homePageDomain.get('location'),
);

const selectContacts = () => createSelector(
  selectHomePageDomain(),
  homePageDomain => homePageDomain.get('contacts'),
);

const selectAge = () => createSelector(
  selectHomePageDomain(),
  homePageDomain => homePageDomain.get('age'),
);

const selectNationality = () => createSelector(
  selectHomePageDomain(),
  homePageDomain => homePageDomain.get('nationality'),
);

const selectAboutMe = () => createSelector(
  selectHomePageDomain(),
  homePageDomain => homePageDomain.get('about'),
);

const selectLanguages = () => createSelector(
  selectHomePageDomain(),
  homePageDomain => homePageDomain.get('languages'),
);

const selectHobbies = () => createSelector(
  selectHomePageDomain(),
  homePageDomain => homePageDomain.get('hobbies').toJS(),
);

export {
  selectDiscovered,
  selectName,
  selectPosition,
  selectLocation,
  selectContacts,
  selectAge,
  selectNationality,
  selectAboutMe,
  selectLanguages,
  selectHobbies,
};
