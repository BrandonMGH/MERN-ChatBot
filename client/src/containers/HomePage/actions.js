import { actionTypes as at } from './constants';

export const discoverHomePageSimpleElement = element => ({
  type: at.DISCOVER_SIMPLE_ELEMENT,
  payload: element,
});

export const discoverHomePageHobbiesElement = element => ({
  type: at.DISCOVER_HOBBIES_ELEMENT,
  payload: element,
});
