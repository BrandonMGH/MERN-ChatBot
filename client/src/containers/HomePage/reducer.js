import { fromJS } from 'immutable';

import { actionTypes as at } from './constants';

const initialState = fromJS({
  discovered: false,
  name: false,
  position: false,
  location: false,
  contactInfo: false,
  age: false,
  nationality: false,
  aboutMe: false,
  languages: false,
  hobbies: fromJS({
    discovered: false,
    programming: false,
    sports: false,
    puzzles: false,
    numismatics: false,
    photography: false,
  }),
});

export default (state = initialState, action) => {
  console.log("HomePage reducer: ", state, action);
  switch (action.type) {
    case at.DISCOVER_HOME_PAGE:
      return state.set('discovered', true);
    case at.DISCOVER_NAME:
      return state.set('name', true);
    case at.DISCOVER_POSITION:
      return state.set('position', true);
    case at.DISCOVER_LOCATION:
      return state.set('location', true);
    case at.DISCOVER_CONTACT_INFO:
      return state.set('contactInfo', true);
    case at.DISCOVER_AGE:
      return state.set('age', true);
    case at.DISCOVER_NATIONALITY:
      return state.set('nationality', true);
    case at.DISCOVER_ABOUT_ME:
      return state.set('aboutMe', true);
    case at.DISCOVER_LANGUAGES:
      return state.set('languages', true);
    case at.DISCOVER_HOBBIES:
      return state.set('hobbies', state.get('hobbies').set('discovered', true));
    default:
      return state;
  }
};
