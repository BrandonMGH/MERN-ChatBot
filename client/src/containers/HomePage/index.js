import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import HomePage from 'components/Pages/Home';

import {
  selectDiscovered, selectName,
  selectPosition, selectLocation,
  selectContactInfo, selectAge,
  selectNationality, selectAboutMe,
  selectLanguages, selectHobbies,
} from './selectors';


const HomePageContainer = props => (
  <HomePage
    discovered={props.discovered}
    name={props.name}
    position={props.position}
    location={props.location}
    contactInfo={props.contactInfo}
    age={props.age}
    nationality={props.nationality}
    aboutMe={props.aboutMe}
    languages={props.languages}
    hobbies={props.hobbies}
  />
);

const mapStateToProps = createStructuredSelector({
  discovered: selectDiscovered(),
  name: selectName(),
  position: selectPosition(),
  location: selectLocation(),
  contactInfo: selectContactInfo(),
  age: selectAge(),
  nationality: selectNationality(),
  aboutMe: selectAboutMe(),
  languages: selectLanguages(),
  hobbies: selectHobbies(),
});

export default connect(mapStateToProps)(HomePageContainer);
