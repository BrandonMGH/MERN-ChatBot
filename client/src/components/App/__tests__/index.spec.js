import React from 'react';
import { create } from 'react-test-renderer';

import HomePage from 'containers/HomePage';
import App, { HomeAdapted } from '../../App';

jest.mock('containers/HomePage', () => 'Home');
jest.mock('../../Pages/Disclaimer', () => 'Disclaimer');
jest.mock('../../Pages/Education', () => 'Education');
jest.mock('../../Pages/Experience', () => 'Experience');
jest.mock('../../Pages/Projects', () => 'Projects');
jest.mock('../../Pages/Feedback', () => 'Feedback');
jest.mock('hoc/layout', () => 'Layout');
jest.mock('react-router-dom', () => ({
  Switch: 'Switch',
  Route: 'Route',
  BrowserRouter: 'BrowserRouter',
}));

const mockComponent = props => (
  <App {...props} />
);

describe('components/<App />', () => {
  it('should render component', () => {
    const tree = create(mockComponent()).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should return correct component', () => {
    const expected = <HomePage />;

    const result = HomeAdapted();

    expect(result).toEqual(expected);
  });
});
