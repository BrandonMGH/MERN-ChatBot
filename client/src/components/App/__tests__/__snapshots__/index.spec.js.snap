// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/<App /> should render component 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      component="Disclaimer"
      exact={true}
      path="/"
    />
    <Layout>
      <Route
        component={[Function]}
        exact={true}
        path="/home"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/experience"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/education"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/projects"
      />
      <Route
        component="Feedback"
        exact={true}
        path="/feedback"
      />
      <Route
        component="Help"
        exact={true}
        path="/help"
      />
    </Layout>
  </Switch>
</BrowserRouter>
`;
